test-api-generator/citrus-test-api-core/src/main/java/org/citrusframework/openapi/testapi/OpenApiParameterFormatter.java

```diff
@@ -58,27 +68,87 @@ private OpenApiParameterFormatter() {
         List<String> values = toList(parameterValue, isObject);
 
-        if (NONE.equals(parameterStyle))  {
-            return parameterName + "="+ (parameterValue != null ? parameterValue.toString() : null);
+        if (NONE.equals(parameterStyle)) {
+            return parameterName + "=" + (parameterValue != null ? parameterValue.toString()
+                : null);
         }
 
         if (DEEPOBJECT.equals(parameterStyle)) {
             return formatDeepObject(parameterName, values);
         }
 
-        FormatParameters formatParameters = determineFormatParameters(parameterName, parameterStyle, explode, isObject);
+        FormatParameters formatParameters = determineFormatParameters(parameterName, parameterStyle,
+            explode, isObject);
 
         if (isObject && explode) {
             return formatParameters.prefix + explode(values, formatParameters.separator);
         } else {
             return formatParameters.prefix + values.stream()
-                    .collect(joining(formatParameters.separator));
+                .collect(joining(formatParameters.separator));
+        }
+    }
+
+    /**
+     * This is an extension to the standard OpenAPI spec, that supports a common coding style for
+     * deeply nested objects, where the json itself is serialized as string.
+     */
+    private static String encodeAsJson(String parameterName, Object parameterValue) {
+        if (parameterValue == null) {
+            return "";
+        }
+
+        JsonNode jsonNode;
+
+        // If it's a string, try parsing as JSON, fallback to URL decoding, as the user might have
+        // encoded the content already.
+        if (parameterValue instanceof String stringValue) {
+            jsonNode = tryParseJson(stringValue);
+            if (jsonNode == null) {
+                String decoded = decode(stringValue, StandardCharsets.UTF_8);
+                jsonNode = tryParseJson(decoded);
+                if (jsonNode == null) {
+                    throw new IllegalArgumentException(format(
+                        "Unable to convert string to JSON. Is this a valid JSON or URL-encoded JSON string?%n%s",
+                        stringValue));
+                }
+            }
+        } else {
+            // If not a string, try to convert object directly
+            jsonNode = objectMapper.valueToTree(parameterValue);
+        }
+
+        try {
+            String compactJson = objectMapper.writeValueAsString(jsonNode).replaceAll("\\R", "");
+
+            // There is no other way to pass this into a query parameter, then URL encoding it.
+            return parameterName + "=" + URLEncoder.encode(compactJson, StandardCharsets.UTF_8);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("Unable to serialize object to JSON string.", e);
+        }
+    }
+
+    private static JsonNode tryParseJson(String input) {
```

Comment:
```suggestion
    private static @Nullable JsonNode tryParseJson(String input) {
```

this helps code scanning tools finding errors.